Source/Private/ASLayoutTransition.mm

```diff
@@ -156,18 +180,40 @@ - (void)calculateSubnodeOperationsIfNeeded
   ASLayout *pendingLayout = _pendingLayout->layout;
 
   if (previousLayout) {
+#if AS_IG_LIST_KIT
+    // IGListDiff completes in linear time O(m+n), so use it if we have it:
+    IGListIndexSetResult *result = IGListDiff(previousLayout.sublayouts, pendingLayout.sublayouts, IGListDiffEquality);
+    _insertedSubnodePositions = findNodesInLayoutAtIndexes(pendingLayout, result.inserts, &_insertedSubnodes);
+    _removedSubnodePositions = findNodesInLayoutAtIndexes(previousLayout, result.deletes, &_removedSubnodes);
+    for (IGListMoveIndex *move in result.moves) {
+      unowned ASDisplayNode *subnode = previousLayout.sublayouts[move.from].layoutElement;
+      _subnodeMoves.push_back(std::pair<ASDisplayNode *, NSUInteger>(subnode, move.to));
```

Comment:
Import `<utility>` and use `std::make_pair(subnode, move.to)`. This:

- Uses type inference to remove code.
- Makes the pair `<unowned ASDisplayNode *, NSUInteger>`, skipping the extra retain/release.

Reply:
Fascinating! So it will infer `unowned`? Is that a general rule for C++ type inference of ObjC types?

Reply:
Yep!